### DISCUSSION TITLE
Equivalent functionality to a Netlify Custom Build Plugin

### DISCUSSION THREAD

**Discussion by gabbibrooks** [vercel/community#889] [Help]
Hi all,

My team is looking into moving deployment platforms and was trying to ensure that Vercel had everything we would need to make a 1-1 move. My question is, is there a way to write a custom build plugin similar to this Netlify build plugin I have that checks the payload data from a deploy hook that we trigger from our CMS to check if we need to rebuild a site based on a documents ID. 
Our CMS will send a payload of information when the hook is triggered including an array of document ids that changed in the hook trigger.

This is how the plugin is setup currently:
```js
module.exports = {
  onPreBuild: ({ utils }) => {
    // document id to check for in payload
    let DOC_ID = 'YkxpFRIAACEAklpS'
    if (
      process.env.INCOMING_HOOK_BODY &&
      process.env.INCOMING_HOOK_BODY.includes(DOC_ID)
    ) {
      utils.build.cancelBuild('No content or code changes - canceling build')
    } else {
      console.log('Content or code changed - continuing build')
    }
  },
}
```

Is there a way to write something similar with a custom integration to do the same process of `Check Deploy Hook Payload -> If document id doesn't exist in deploy payload -> Cancel Build`?

**Comment by amyegan** [comment]
Hi @brooksztb! I'm not familiar with that feature on Netlify, but I'm looking into it and checking with the team. I want to make sure I don't mistakenly give you the wrong answer on this.
